src/AI/WallyAI.java

```diff
@@ -918,14 +922,30 @@ private static int valueUnit(Unit unit, Location locale, boolean includeCurrentH
   {
     int value = unit.model.getCost();
 
+    if( unit.CO.isEnemy(locale.getOwner()) &&
+            unit.model.hasActionType(UnitActionFactory.CAPTURE)
+            && locale.isCaptureable() )
+      value += valueTerrain(unit.CO, locale.getEnvironment().terrainType); // Strongly value units that threaten capture
     if( includeCurrentHealth )
       value *= unit.getHP();
-    value *= (unit.model.hasActionType(UnitActionFactory.CAPTURE) && locale.isCaptureable()) ? 8 : 1; // Strongly value capturing units
     value -= locale.getEnvironment().terrainType.getDefLevel(); // Value things on lower terrain more, so we wall for equal units if we can get on better terrain
 
     return value;
   }
 
+  private static int valueTerrain(Commander co, TerrainType terrain)
+  {
+    int value = 0;
+    if( terrain.isProfitable() )
+      value += co.gameRules.incomePerCity * TERRAIN_FUNDS_WEIGHT;
+    if( co.unitProductionByTerrain.containsKey(terrain) )
+      value += TERRAIN_INDUSTRY_WEIGHT;
+    if( TerrainType.HEADQUARTERS == terrain
+        || TerrainType.LAB == terrain )
+      value += TERRAIN_HQ_WEIGHT;
```

Comment:
In a high-funds game, the HQ might be undervalued. We allow up to 20K, which with the 2.5 multiplier makes a standard city worth more than HQ.

Reply:
It's additive.

Reply:
Ah. I guess if I actually used those squishy white orbs in my face I would have known that.